enforcer/src/keystate/keystate_import_cmd.c

```diff
@@ -0,0 +1,467 @@
+        key_state_free(key_state);
+        hsm_destroy_context(hsm_ctx);
+        return -1;
+    }
+
+    if (!(key_state = key_state_new(dbconn))
+                || key_state_set_key_data_id(key_state, keydataid)
+                || key_state_set_type(key_state, KEY_STATE_TYPE_RRSIGDNSKEY)
+                || key_state_set_last_change (key_state, time)
+                || key_state_set_ttl (key_state, policy_keys_ttl(policy))
+                || key_state_set_state(key_state, keytype == 1 ? ksk_mapping[keystate][2] : zsk_mapping[keystate][2])
+                || key_state_create(key_state))
+    {
+        ods_log_error("[%s] key state creation for RRSIGDNSKEY failed, database or memory error", module_str);
+        key_state_free(key_state);
+        hsm_destroy_context(hsm_ctx);
+        return -1;
+    }
+
+    ttl = max(min(policy_zone_soa_ttl(policy), policy_zone_soa_minimum(policy)),
+            (policy_denial_type(policy) == POLICY_DENIAL_TYPE_NSEC3
+                ? ( policy_denial_ttl(policy) > policy_signatures_max_zone_ttl(policy)
+                    ? policy_denial_ttl(policy)
+                    : policy_signatures_max_zone_ttl(policy))
+                : policy_signatures_max_zone_ttl(policy)));
+
+    if (!(key_state = key_state_new(dbconn))
+                || key_state_set_key_data_id(key_state, keydataid)
+       /*         || hsm_key_set_backup(hsm_key, (hsm->require_backup ? HSM_KEY_BACKUP_BACKUP_REQUIRED : HSM_KEY_BACKUP_NO_BACKUP))*/
+                || key_state_set_type(key_state, KEY_STATE_TYPE_RRSIG)
+                || key_state_set_last_change (key_state, time)
+                || key_state_set_minimize(key_state, key_data_minimize(key) & 1)
+                || key_state_set_ttl (key_state, ttl)
+                || key_state_set_state(key_state, keytype == 1 ? ksk_mapping[keystate][3] : zsk_mapping[keystate][3])
+                || key_state_create(key_state))
+    {
+        ods_log_error("[%s] key state creation for RRSIG failed, database or memory error", module_str);
+        key_state_free(key_state);
+        hsm_destroy_context(hsm_ctx);
+        return -1;
+    }
+    ods_log_debug("[%s] key state with this locator %s is created successfully", module_str, ckaid);
+
+    key_state_free(key_state);
+    key_data_free(key);
+    hsm_destroy_context(hsm_ctx);
+    policy_free(policy);
+    return 0;
+}
+
+
+
+static void
+usage(int sockfd)
+{
+    client_printf(sockfd,
+		"key import             Import DNSKEY(s) for a given zone to the database.\n"
+		"      --cka_id <CKA_ID>		(aka -k)  cka_id\n"
+		"      --repository <repository>	(aka -r) repository\n"
+		"      --zone <zone>			(aka -z)  zone \n"
+		"      --bits <size>			(aka -b) bits\n"
+		"      --algorithm <algorithm>		(aka -g) algorithm \n"
+		"      --keystate <state>		(aka -e) keystate\n"
+		"      --keytype <type>			(aka -t) keytype\n"
+		"      --time <time>			(aka -w) time\n"
```

Comment:
thanks for review, I used 1.4 format. inception time? is it good or have any suggestion?